user-guide/howto/files/pxe/dnsmasq.conf

```diff
@@ -0,0 +1,55 @@
+# Configuration file for dnsmasq.
+# Disable DNS
+port=0
+
+# Set the domain for dnsmasq. this is optional, but if it is set, it
+# does the following things.
+# 1) Allows DHCP hosts to have fully qualified domain names, as long
+#     as the domain part matches this setting.
+# 2) Sets the "domain" DHCP option thereby potentially setting the
+#    domain of all systems configured by DHCP
+# 3) Provides the domain part for "expand-hosts"
+domain=test.example.com
+
+# Uncomment this to enable the integrated DHCP server, you need
+# to supply the range of addresses available for lease and optionally
+# a lease time. If you have more than one network, you will need to
+# repeat this for each network on which you want to supply DHCP
+# service.
+dhcp-range=set:nodes,10.10.0.50,10.10.0.100,255.255.255.0,12h
+
+# Always give the host with Ethernet address 11:22:33:44:55:66
+# the name node99 and tag node99 and IP address 10.0.60.60
+#dhcp-host=set:node99,11:22:33:44:55:66,node99,10.10.0.200,24h
+
+# Override the default route supplied by dnsmasq, which assumes the
+# router is the same machine as the one running dnsmasq.
+dhcp-option=option:router,10.10.0.1
+
+# Set the DNS server address
+dhcp-option=option:dns-server,10.10.0.4
+
+# Boot for Etherboot gPXE. The idea is to send two different
+# filenames, the first loads gPXE, and the second tells gPXE what to
+# load. The dhcp-match sets the gpxe tag for requests from gPXE.
+dhcp-match=set:gpxe,175 # gPXE sends a 175 option.
+dhcp-boot=tag:!gpxe,undionly.kpxe
+dhcp-boot=menu.ipxe
+#dhcp-boot=tag:nodes,menu.ipxe
+
+# Run custom iPXE script on specific server with tag "node99"
+#dhcp-boot=tag:node99,esdc-latest.ipxe
```

Comment:
I recommend separating this custom node config lines to a separate file. It can be easily altered if necessary.
E.g: ```conf-dir=/opt/local/etc/dnsmasq.d```